src/dac.rs

```
@@ -11,306 +11,147 @@ use super::{
+                        None,
+                        trigger_config,
+                    );
+
+                Self {
+                    transfer,
+                    // Note(unsafe): This buffer is only used once and provided for the next DMA transfer.
+                    next_buffer: unsafe { Some(&mut DAC_BUF[$index][1]) },
+                    first_transfer: true,
+                }
+            }
+
+            /// Acquire the next output buffer to populate it with DAC codes.
+            pub fn acquire_buffer(
+                &mut self,
+            ) -> &'static mut [u16; SAMPLE_BUFFER_SIZE] {
+                self.next_buffer.take().unwrap()
+            }
+
+            /// Enqueue the next buffer for transmission to the DAC.
+            ///
+            /// # Args
+            /// * `data` - The next data to write to the DAC.
+            pub fn release_buffer(
+                &mut self,
+                next_buffer: &'static mut [u16; SAMPLE_BUFFER_SIZE],
+            ) {
+                // If the last transfer was not complete, we didn't write all our previous DAC codes.
+                // Wait for all the DAC codes to get written as well.
+                if self.first_transfer {
+                    self.first_transfer = false
+                } else {
+                    // Note: If a device hangs up, check that this conditional is passing correctly, as
+                    // there is no time-out checks here in the interest of execution speed.
+                    while !self.transfer.get_transfer_complete_flag() {}
+                }
+
+                // Start the next transfer.
+                self.transfer.clear_interrupts();
+                let (prev_buffer, _) =
+                    self.transfer.next_transfer(next_buffer).unwrap();
+
+                // .unwrap_none() https://github.com/rust-lang/rust/issues/62633
+                self.next_buffer.replace(prev_buffer);
```

I think it's safe to silently ignore the return value here - why add documentation around the unwrap?

---

There wasn't any documentation.
This makes it clear that we are expecting a `None` in return by design. And we can enforce it once that is stabilized.